scripts/basic_twoword_classifier.py

```diff
@@ -0,0 +1,30 @@
+import torch.nn as nn
+import torch.nn.functional as F
+import scripts.composition_functions as comp_functions
+
+
+class BasicTwoWordClassfier(nn.Module):
+    """
+    this class includes a basic two-word classifier with one hidden layer and one output layer.
+    :param input_dim: the dimension of the input vector where only embedding size (2 times the size of the embedding of
+    a single word vector) is needed, batch size is implicit
+    :param hidden_dim : the dimension of the hidden layer, batch size is implicit
+    :param label_nr : the dimension of the output layer, i.e. the number of labels
+    """
+
+    def __init__(self, input_dim, hidden_dim, label_nr):
+        super(BasicTwoWordClassfier, self).__init__()
+        self.hidden_layer = nn.Linear(input_dim, hidden_dim)
+        self.output_layer = nn.Linear(hidden_dim, label_nr)
```

Comment:
self.hidden_layer = nn.Linear(input_dim, hidden_dim)
        self.output_layer = nn.Linear(hidden_dim, label_nr)

it is not necessary for the code to work but because we are defining a real class now I would prefer class variables to be protected. this is done with self._hidden_layer .. you can then define them as class properties with the following:
```
@property
def hidden_layer(self):
     return self._hidden_layer

```

Reply:
It is a bit confusing how to do object oriented programming with Python and a lot of things will just work one or the other way because python. but it is more of a conventional thing to make class instance variables "protected".
I found this short summary useful:
https://www.tutorialsteacher.com/python/private-and-protected-access-modifiers-in-python